app/assets/javascripts/actions/revisions_actions.js

```diff
@@ -18,15 +21,16 @@ import { sortRevisionsByDate } from '../utils/revision_utils';
 import { INCREASE_LIMIT } from '../constants/revisions';
 import { STUDENT_ROLE } from '../constants/user_roles';
 
-const fetchCourseScopedRevisionsPromise = async (course, limit) => {
-  const response = await request(`/courses/${course.slug}/revisions.json?limit=${limit}&course_scoped=true`);
+const fetchAllArticles = async (course) => {
+  const response = await request(`/courses/${course.slug}/articles.json?limit=500`);
```

Comment:
I think this limit is inevitably going to be confusing in some edge cases with large numbers of articles.

I think perhaps the solution could be to disable course-specific edits altogether for courses with too many articles to safely do this. My suggestion is to keep the limit, but don't show the `Course Specific` button at all if the course's article count is over 500 (which you can check via the `course` object without needing to query articles.json).

Reply:
Hmm I guess that's one way to do it. Do you think the limit should be increased though (from the current 500)?

Reply:
I think that limit is fine.